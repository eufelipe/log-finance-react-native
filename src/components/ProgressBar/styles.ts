import styled from 'styled-components/native';

const patternHexColor = /^#([0-9a-f]{3}){1,2}$/i;

export const Container = styled.View`
  background: ${({theme}) => theme.colors.accent};
  border-radius: ${({theme}) => theme.spacing.small}px;
`;

export const Progress = styled.View<{percent?: number; color?: string}>`
  background: ${({theme, color}) =>
    color && patternHexColor.test(color) ? color : theme.colors.seconday}50;
  border-radius: ${({theme}) => theme.spacing.small}px;
  width: ${({percent = 0}) => (percent > 100 ? 100 : percent)}%;
  height: 10px;
`;
export const Title = styled.Text`
  font-size: 16px;
`;
