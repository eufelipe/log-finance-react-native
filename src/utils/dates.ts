import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
