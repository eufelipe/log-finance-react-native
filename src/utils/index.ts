export {default as number} from './number';
export {default as strings} from './strings';
