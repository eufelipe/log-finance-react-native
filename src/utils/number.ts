export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
