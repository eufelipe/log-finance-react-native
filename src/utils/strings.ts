export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
