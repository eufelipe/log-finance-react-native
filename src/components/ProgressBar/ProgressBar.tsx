import React from 'react';
import {Container, Progress} from './styles';

interface ProgressBarProps {
  percent: number;
  color?: string;
}

const ProgressBar = ({percent, color}: ProgressBarProps): JSX.Element => {
  return (
    <Container>
      <Progress percent={percent} color={color} />
    </Container>
  );
};

export default ProgressBar;
