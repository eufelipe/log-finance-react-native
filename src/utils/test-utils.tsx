/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

import {NavigationContainer} from '@react-navigation/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
