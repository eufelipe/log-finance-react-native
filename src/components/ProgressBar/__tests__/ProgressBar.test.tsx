/* eslint-disable @typescript-eslint/unbound-method */
/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render} from 'utils/test-utils';
import ProgressBar from '../ProgressBar';

describe('ProgressBar', () => {
  it('Test match snapshot ProgressBar', () => {
    // given
    const props = {
      percent: 10,
    };

    // when
    const {toJSON} = render(<ProgressBar {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  it('Test match snapshot ProgressBar with color', () => {
    // given
    const props = {
      percent: 45,
      color: "#0000000"
    };

    // when
    const {toJSON} = render(<ProgressBar {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
