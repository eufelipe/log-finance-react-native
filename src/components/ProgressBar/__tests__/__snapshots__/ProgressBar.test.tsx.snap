// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Test match snapshot ProgressBar 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#181D26",
        "borderRadius": 10.714285714285714,
      },
    ]
  }
>
  <View
    percent={10}
    style={
      Array [
        Object {
          "backgroundColor": "#37AF7850",
          "borderRadius": 10.714285714285714,
          "height": 10,
          "width": "10%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Test match snapshot ProgressBar with color 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#181D26",
        "borderRadius": 10.714285714285714,
      },
    ]
  }
>
  <View
    color="#0000000"
    percent={45}
    style={
      Array [
        Object {
          "backgroundColor": "#37AF7850",
          "borderRadius": 10.714285714285714,
          "height": 10,
          "width": "45%",
        },
      ]
    }
  />
</View>
`;
